runtime/src/main/java/io/quarkiverse/openapi/generator/providers/ApiKeyAuthenticationProvider.java

```diff
@@ -44,7 +45,8 @@ public void filter(ClientRequestContext requestContext) throws IOException {
                 requestContext.setUri(UriBuilder.fromUri(requestContext.getUri()).queryParam(apiKeyName, getApiKey()).build());
                 break;
             case cookie:
-                requestContext.getCookies().put(apiKeyName, new Cookie.Builder(apiKeyName).value(getApiKey()).build());
+                final Cookie cookie = new Cookie.Builder(apiKeyName).value(getApiKey()).build();
+                requestContext.getHeaders().add(HttpHeaders.COOKIE, cookie);
```

Comment:
Just keep one line here
```suggestion
                requestContext.getHeaders().add(HttpHeaders.COOKIE, new Cookie.Builder(apiKeyName).value(getApiKey()).build());
```

Reply:
Sure, applied.